extensions/terminal-suggest/src/terminalSuggestMain.test.ts

```diff
@@ -7,63 +7,65 @@ import { deepStrictEqual, strictEqual } from 'assert';
-		createTestCase('code --extensions-dir |', [], 'folders', availableSpecs);
-		createTestCase('code --list-extensions |', codeOptions, 'neither', availableSpecs);
-		createTestCase('code --show-versions |', codeOptions, 'neither', availableSpecs);
-		createTestCase('code --category |', ['azure', 'data science', 'debuggers', 'extension packs', 'education', 'formatters', 'keymaps', 'language packs', 'linters', 'machine learning', 'notebooks', 'programming languages', 'scm providers', 'snippets', 'testing', 'themes', 'visualization', 'other'], 'neither', availableSpecs);
-		createTestCase('code --category a|', ['azure'], 'neither', availableSpecs);
-		createTestCase('code-insiders --list-extensions |', codeOptions, 'neither', availableSpecs);
-		createTestCase('code-insiders --show-versions |', codeOptions, 'neither', availableSpecs);
-		createTestCase('code-insiders --category |', ['azure', 'data science', 'debuggers', 'extension packs', 'education', 'formatters', 'keymaps', 'language packs', 'linters', 'machine learning', 'notebooks', 'programming languages', 'scm providers', 'snippets', 'testing', 'themes', 'visualization', 'other'], 'neither', availableSpecs);
-		createTestCase('code-insiders --category a|', ['azure'], 'neither', availableSpecs);
-		createTestCase('code-insiders --category azure |', [], 'neither', availableSpecs);
-	});
-	suite('Cursor not at the end of the line', () => {
-		createTestCase('code | --locale', codeOptions, 'neither', availableSpecs);
-		createTestCase('code --locale | && ls', ['bg', 'de', 'en', 'es', 'fr', 'hu', 'it', 'ja', 'ko', 'pt-br', 'ru', 'tr', 'zh-CN', 'zh-TW'], 'neither', availableSpecs);
-		createTestCase('code-insiders | --locale', codeOptions, 'neither', availableSpecs);
-		createTestCase('code-insiders --locale | && ls', ['bg', 'de', 'en', 'es', 'fr', 'hu', 'it', 'ja', 'ko', 'pt-br', 'ru', 'tr', 'zh-CN', 'zh-TW'], 'neither', availableSpecs);
-	});
+interface ITestSpec {
+	input: string;
+	expectedCompletionLabels: string[];
+	resourcesRequested: 'files' | 'folders' | 'both' | 'neither';
+}
 
-	function createTestCase(commandLineWithCursor: string, expectedCompletionLabels: string[], resourcesRequested: 'files' | 'folders' | 'both' | 'neither', availableSpecs: Fig.Spec[]): void {
-		const commandLine = commandLineWithCursor.split('|')[0];
-		const cursorPosition = commandLineWithCursor.indexOf('|');
-		const prefix = commandLine.slice(0, cursorPosition).split(' ').pop() || '';
-		const filesRequested = resourcesRequested === 'files' || resourcesRequested === 'both';
-		const foldersRequested = resourcesRequested === 'folders' || resourcesRequested === 'both';
-		test(commandLineWithCursor, function () {
+suite('Terminal Suggest', () => {
+	for (const testSpec of testSpecs) {
+		test(testSpec.input, () => {
+			const commandLine = testSpec.input.split('|')[0];
+			const cursorPosition = testSpec.input.indexOf('|');
+			const prefix = commandLine.slice(0, cursorPosition).split(' ').pop() || '';
```

Comment:
nit: `.at(-1)` instead of `pop()`? `pop` says to me that it's important that the element is leaving the original array